### ISSUE TITLE
Collate function on SparseTensor

### ISSUE THREAD

**Issue by rubenwiersma** [pyg-team/pytorch_geometric#3140]
## 🐛 Bug

If I add a `SparseTensor` element to a `Data` object, the collate function throws a type error.

## To Reproduce

1. Create a transform that adds a `SparseTensor` to a `Data` object, e.g.: `data.sparse = SparseTensor.eye(5)`
2. Use the minibatcher to load the data.
3. Error thrown at https://github.com/pyg-team/pytorch_geometric/blob/19276d17d8af5c8b37dc6f63a7198d93564b6c61/torch_geometric/data/collate.py#L155
```
TypeError: 'int' object is not iterable
```

## Expected behavior
The line should be:
```python
        cat_dims = (cat_dim,) if isinstance(cat_dim, int) else cat_dim
```

On a sidenote, is there a specific reason diagonal stacking is not the default for SparseTensors?

**Comment by rusty1s** [comment]
Thank you! I fixed it in master. We are only doing diagonal stacking for adjacency matrices (`adj*`) to allow for correct default batching for other sparse variants, such as sparse node feature matrices.

**Comment by rusty1s** [comment]
I just released `torch-geometric==2.0.1` that fixes this issue :)
